### PR TITLE
Get Harvest by URL

### PR DESCRIPTION
**Pull request by avdata99**
Fix error when we check for a Harvest source by URL and not for ID

### CODE REVIEW

**Comment by avdata99** [comment]
This is need in order to pass CircleCI tests at [GeoDataGov PR](https://github.com/GSA/ckanext-geodatagov/pull/120) @adborden. I have no permissions to assign this to you.

File: ckanext/harvest/logic/action/get.py [hunk 1]
```diff
@@ -30,11 +30,23 @@ def harvest_source_show(context,data_dict):
...
+    :param url: url of the harvest source (as an alternative to the id)
+    :type url: string
 
     :returns: harvest source metadata
     :rtype: dictionary
     '''
+    model = context.get('model')
 
+    # Find the source by URL
+    if data_dict.get('url') and not data_dict.get('id'):
```
Comment (adborden) [inline ckanext/harvest/logic/action/get.py:43]: Are these standard python `dict`s? Will this throw an error if the key does not exist, or are they guaranteed to exist?

**Comment by adborden** [comment]
When is it possible that a harvest source would have a URL but not an Id?

**Comment by adborden** [comment]
Oh, I misunderstood this comment https://github.com/GSA/datagov-ckan-multi/issues/153#issuecomment-570586545. This is pulling a fix from upstream.

**Pull request merged**